### DISCUSSION TITLE
进行OCR处理时报错，异常状态码：102，报错文档见附件

### DISCUSSION THREAD

**Discussion by nononamezdm** [hiroi-sora/Umi-OCR#558] [Q&A] [answered]
异常状态码：102
异常信息：[Error] Doc P0
[Error] OCR code:904 msg:识别器输出值反序列化JSON失败。异常信息：[Expecting value: line 2 column 1 (char 2)]。原始内容：[
]

异常状态码：102
异常信息：[Error] Doc P1
[Error] OCR code:902 msg:向识别器进程传入指令失败，疑似子进程已崩溃。[Errno 32] Broken pipe

[UMI-OCR报错文档.pdf](https://github.com/user-attachments/files/15786500/UMI-OCR.pdf)

**Comment by hiroi-sora** [comment]
你好，我测试了你的PDF，在我这边是正常的。

![image](https://github.com/hiroi-sora/Umi-OCR/assets/56373419/77a707f9-0183-425b-85f5-5fbd55f63426)

请问，进行别的OCR任务（比如截图、jpg/png批量图片识别）正常吗？是否只有文档识别会报错？

别的PDF文件会报错吗，还是只有你上传的这一个样例有问题？

**Reply by nononamezdm** [reply]
谢谢回复，不是所有文档都报错，但是也有别的文档同样报错；截图OCR没问题。

**Reply by nononamezdm** [reply]
这个文档提示识别完成，但实际上什么也没有识别出来。
[Q.pdf](https://github.com/user-attachments/files/15788085/Q.pdf)

**Reply by hiroi-sora** [reply]
Q.pdf 这份文档确实很奇怪，它的数据流里既没有图片，又没有字符。但是放大之后，文字的清晰度不变，那么肯定不是像素图。我怀疑这文档将所有字符转换为了矢量图。

不过，Umi-OCR 依然可以处理这类非标准文档：设置→文档处理→内容提取模式→改为 **整页强制OCR** 即可。

别的报错的文档，也可以试试这个选项，兼容性比较好。

![image](https://github.com/hiroi-sora/Umi-OCR/assets/56373419/a926c0d9-1f3a-4eb2-9882-e7c669a76cdd)

**Reply by nononamezdm** [reply]
强制OCR试了，一直处于无进展的状态。不知道是什么问题，两台电脑都是这样。
![Q](https://github.com/hiroi-sora/Umi-OCR/assets/55967118/f90d11fb-37dd-48b0-bde7-7819a140e99a)

**Reply by hiroi-sora** [reply]
可能你下载的是 v2.1.2 刚发布时候的包，存在相关Bug，现在已修复。请重新下载 v2.1.2 试试。

https://hiroi-sora.lanzoul.com/s/umi-ocr
https://github.com/hiroi-sora/Umi-OCR/releases/latest

**Reply by nononamezdm** [reply]
重新下载了，至少我发的文档可以处理了，多谢！

**Comment by nononamezdm** [comment]
你好！目前识别的问题基本解决了。还有一个问题是批处理文档的时候，经常会出现异常：异常状态码：102异常信息：[Error] Doc P0[Error] OCR code:904 msg:识别器输出值反序列化JSON失败。异常信息：[Expecting value: line 2 column 1 (char 2)]。原始内容：[]
实际上批处理的文件和页数都不算多，有一次一共不到10页的2个文档，在处理到第一个文档的第4页就开始报异常，然后就一直异常，知道所有文档都处理完。但退出程序后重启再处理，报异常的文档能够正常处理，但批处理的问题仍然存在。

----- 原始邮件 -----
发件人：hiroi-sora ***@***.***>
收件人：hiroi-sora/Umi-OCR ***@***.***>
抄送人：nononamezdm ***@***.***>,  Author ***@***.***>
主题：Re:_[hiroi-sora/Umi-OCR]_进行OCR处理时报错，异常状态码：102，报错文档见附件_(Discussion_#558)
日期：2024年06月11日 21点34分


可能你下载的是 v2.1.2 刚发布时候的包，存在相关Bug，现在已修复。请重新下载 v2.1.2 试试。
https://hiroi-sora.lanzoul.com/s/umi-ocr

https://github.com/hiroi-sora/Umi-OCR/releases/latest

**Reply by hiroi-sora** [reply, marked as answer]
使用“整页强制OCR”模式也是容易报错102吗？

你当前用的是 Paddle 还是 Rapid 版本？尝试换用另一个版本看看，如果用的是Paddle，就换成Rapid试试

**Comment by nononamezdm** [comment]
我用的是paddle版，关掉360天擎以后暂时没出现这个错误。有问题再请教，谢谢！
----- 原始邮件 -----
发件人：hiroi-sora ***@***.***>
收件人：hiroi-sora/Umi-OCR ***@***.***>
抄送人：nononamezdm ***@***.***>,  Author ***@***.***>
主题：Re:_[hiroi-sora/Umi-OCR]_进行OCR处理时报错，异常状态码：102，报错文档见附件_(Discussion_#558)
日期：2024年06月12日 12点10分


使用“整页强制OCR”模式也是容易报错102吗？
你当前用的是 Paddle 还是 Rapid 版本？尝试换用另一个版本看看，如果用的是Paddle，就换成Rapid试试

**Comment by nononamezdm** [comment]
你好！到目前遇到附件里的3个文档处理时报错：异常状态码：102异常信息：[Error] Doc P0[Error] OCR code:904 msg:识别器输出值反序列化JSON失败。异常信息：[Expecting value: line 2 column 1 (char 2)]。原始内容：[]
异常状态码：102异常信息：[Error] Doc P1[Error] OCR code:902 msg:向识别器进程传入指令失败，疑似子进程已崩溃。[Errno 32] Broken pipe
谢谢！
----- 原始邮件 -----
发件人：hiroi-sora ***@***.***>
收件人：hiroi-sora/Umi-OCR ***@***.***>
抄送人：nononamezdm ***@***.***>,  Author ***@***.***>
主题：Re:_[hiroi-sora/Umi-OCR]_进行OCR处理时报错，异常状态码：102，报错文档见附件_(Discussion_#558)
日期：2024年06月12日 12点10分


使用“整页强制OCR”模式也是容易报错102吗？
你当前用的是 Paddle 还是 Rapid 版本？尝试换用另一个版本看看，如果用的是Paddle，就换成Rapid试试

**Reply by nononamezdm** [reply]
抱歉，以为回复过了￣□￣｜｜ 后来换成Rapid版就没再出现问题。
